src/hooks/useSwapStats.ts

```diff
@@ -0,0 +1,85 @@
+import { BLOCK_TIME, ChainId } from "../constants"
+import { useCallback } from "react"
+import { useQuery } from "@tanstack/react-query"
+
+const swapStatsURI = "https://ipfs.saddle.exchange/swap-stats.json"
+
+type PoolStats = {
+  oneDayVolume: string
+  apy: string
+  tvl: string
+  utilization: string
+}
+type SwapStats = {
+  [chainId in ChainId]?: Partial<{
+    [swapAddress: string]: PoolStats
+  }>
+}
+
+type SwapStatsReponse = {
+  [chainId in ChainId]?: {
+    [swapAddress: string]: {
+      oneDayVolume: number
+      APY: number
+      TVL: number
+    }
+  }
+}
+export function useSwapStats() {
+  return useQuery(
+    ["swapStats"],
+    async () => {
+      const res = await fetch(`${swapStatsURI}`)
+      if (res.status >= 200 && res.status < 300) {
+        return res.json()
+      }
+      throw new Error("Unable to fetch swap stats from IPFS")
+    },
+    {
+      select: useCallback(
```

Comment:
Why is this a `useCallback` instead of a plain function?

Reply:
https://github.com/TkDodo/blog-comments/discussions/57#discussioncomment-3759612
> we do memoize the result of select, but only if:
> 
> - data hasn't changed
> 
> - the functional identity hasn't changed
> 
> the first one should be obvious - select needs to re-run if we get new data from the server.
> the second one is there to ensure that if you closure over dependencies, you also get a correct result. For example, if you select an item from a list, and the id of that item comes in as a prop, we need to re-run select when that prop changes. That's where useCallback helps, as I've shown in the example: The functional identity will change only when id changes, so we know when to re-run select.